octoprint_slicer/static/js/slicer.js

```diff
@@ -424,36 +430,64 @@ $(function() {
+            for(var i = self.walls.children.length - 1; i >= 0; i--) {
+                var obj = self.walls.children[i];
+                self.walls.remove(obj);
+            }
 
-            segments = segments || 20;
-            var geometry = new THREE.PlaneGeometry(width, depth, segments, segments);
-            var materialEven = new THREE.MeshBasicMaterial({color: 0xccccfc});
-            var materialOdd = new THREE.MeshBasicMaterial({color: 0x444464});
-            var materials = [materialEven, materialOdd];
-            for (var x = 0; x < segments; x++) {
-                for (var y = 0; y < segments; y++) {
-                    var i = x * segments + y;
-                    var j = 2 * i;
-                    geometry.faces[ j ].materialIndex = geometry.faces[ j + 1 ].materialIndex = (x + y) % 2;
-                }
+            var segments = self.BEDSIZE_X_MM / 10;
+            var bedRadius = self.BEDSIZE_X_MM / 2;
+            var geometry = new THREE.CircleGeometry(bedRadius, 60);
+            var texture = new CheckerBoardTexture(0xccccfc, 0x444464, segments, segments);
+            var circle = new THREE.Mesh(geometry, new THREE.MeshPhongMaterial({ map: texture, transparent: true, opacity: 0.5, side: THREE.DoubleSide }));
+            self.floor.add(circle);
+
+            var cylGeometry = new THREE.CylinderGeometry(bedRadius, bedRadius, self.BEDSIZE_Z_MM, segments, self.BEDSIZE_Z_MM, true);
+            //This material will only make the inside of the cylinder walls visible while allowing the outside to be transparent.
+            var material = new THREE.MeshBasicMaterial({ color: 0x8888fc, side: THREE.BackSide, transparent: true, opacity: 0.5 });
+            // Move the walls up to the floor
+            cylGeometry.applyMatrix(new THREE.Matrix4().makeTranslation(0, self.BEDSIZE_Z_MM / 2, 0));
+            var wall = new THREE.Mesh(cylGeometry, material);
+            //rotate the walls so they are upright
+            wall.rotation.x = Math.PI/2;
+            self.walls.add(wall);
+
+            //Add text to indicate front of print bed
+            (new THREE.FontLoader()).load( PLUGIN_BASEURL + "slicer/static/js/optimer_bold.typeface.json", function ( font ) {
+                self.createText(font, "Front", bedRadius * 2, bedRadius * 2, self.floor);
+            } );
+        }
+
+        self.drawBedFloor = function ( width, depth, segments ) {
+            for(var i = self.floor.children.length - 1; i >= 0; i--) {
+                var obj = self.floor.children[i];
+                self.floor.remove(obj);
             }
-            var mesh = new THREE.Mesh(geometry, new THREE.MeshFaceMaterial(materials));
+
+            var xSegments = segments || self.BEDSIZE_X_MM / 10;
+            var ySegments = segments || self.BEDSIZE_Y_MM / 10;
+            var geometry = new THREE.PlaneGeometry(width, depth, xSegments, ySegments);
+            var texture = new CheckerBoardTexture(0xccccfc, 0x444464, xSegments, ySegments);
```

Comment:
I do notice the lag, let me look at that a little closer, One thing to do is drop the number of mesh segments since the texture does right thing now.